src/qmapshack/gis/prj/IGisProject.cpp

```diff
@@ -1141,6 +1147,22 @@ void IGisProject::sortItems()
     }
 
     addChildren(items);
+    CGisListWks* widget = dynamic_cast<CGisListWks*>(treeWidget());
+    if(widget != nullptr)
+    {
+        for(QTreeWidgetItem * searchItem : others)
+        {
+            if(searchItem->text(0) == tr("Search:"))
+            {
+                CSearchLineEdit* edit = new CSearchLineEdit(widget, this, searchItem);
+                edit->setText(projectSearch.getSearchText());
+                //eColumnName is 1
+                widget->setItemWidget(searchItem, 1, edit);
```

Comment:
Same with this. Especially the fact that I can't use the enum, since that would require a circular inclusion

Reply:
Why don't you handle the filter item the same way as all other item? I don't really see why this shouldn't be possible.